### PR TITLE
Declare material models and specs

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/models/course/material.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Course::Material < ActiveRecord::Base
+  belongs_to :creator, inverse_of: :files, class_name: User.name
```
Comment (lowjoel) [inline app/models/course/material.rb]: I don't think the user should be linked to every file. ActiveRecord-Userstamp already creates the belongs_to association.

File: app/models/course/material.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Course::Material < ActiveRecord::Base
```
Comment (lowjoel) [inline app/models/course/material.rb]: Maybe Material should be a namespace, call this Course::Material::File?

Comment (allenwq) [inline app/models/course/material.rb]: This was the name I was using before exactly, but got a name conflict with global `File`.

Comment (lowjoel) [inline app/models/course/material.rb]: okay, leave it then.

File: app/models/course/material.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  belongs_to :folder, inverse_of: :materials, class_name: Course::Material::Folder.name
+  acts_as_attachable
```
Comment (lowjoel) [inline app/models/course/material.rb]: oops, sorry. read contributing.md on the order.

File: db/migrate/20150803065716_create_materials.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class CreateMaterials < ActiveRecord::Migration
+  def change
+    create_table :course_material_folders do |t|
+      t.references :parent_folder, foreign_key: { references: :course_material_folders }
+      t.references :owner, polymorphic: true
```
Comment (lowjoel) [inline db/migrate/20150803065716_create_materials.rb]: add an index

File: app/models/course/material.rb (hunk 3 above)
Comment (allenwq) [inline app/models/course/material.rb]: ah, my fault Thanks !

File: db/migrate/20150803065716_create_materials.rb (hunk 4 above)
Comment (lowjoel) [inline db/migrate/20150803065716_create_materials.rb]: wait, no need, automatic.

File: db/migrate/20150803065716_create_materials.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      t.string :name, null: false
+      t.text :description
+      t.datetime :start_at, null: false
```
Comment (lowjoel) [inline db/migrate/20150803065716_create_materials.rb]: why is this null false?

Comment (allenwq) [inline db/migrate/20150803065716_create_materials.rb]: I think every folder should have a start time ?  And if you want the folder to always be available, just set the start time earlier than current time.

**Pull request merged**